libeos-payg/manager.c

```diff
@@ -1166,30 +1360,20 @@ epg_manager_save_state_async (EpgProvider         *provider,
 
   g_autoptr(GTask) task = g_task_new (self, cancellable, callback, user_data);
   g_task_set_source_tag (task, epg_manager_save_state_async);
-  epg_multi_task_attach (task, 3);
-
-  /* Save the expiry time. */
-  g_autoptr(GFile) expiry_time_file = get_expiry_time_file (self);
-
-  union
-    {
-      guint64 u64;
-      const guint8 u8[8];
-    } expiry_time_secs;
-  G_STATIC_ASSERT (sizeof (expiry_time_secs.u8) == sizeof (self->expiry_time_secs));
-  expiry_time_secs.u64 = self->expiry_time_secs;
+  epg_multi_task_attach (task, 4);
 
-  g_autoptr(GBytes) expiry_time_bytes = g_bytes_new (expiry_time_secs.u8,
-                                                     sizeof (expiry_time_secs.u8));
+  /* Save the wall clock time. */
+  g_autoptr(GFile) wallclock_time_file = get_wallclock_time_file (self);
+  guint64 wallclock_seconds = epg_clock_get_wallclock_time (self->clock);
+  write_guint64_to_file (wallclock_seconds, wallclock_time_file, task, cancellable);
 
-  g_file_replace_contents_bytes_async (expiry_time_file,
-                                       expiry_time_bytes,
-                                       NULL,  /* ETag */
-                                       FALSE,  /* no backup */
-                                       G_FILE_CREATE_PRIVATE,
-                                       cancellable,
-                                       file_replace_cb,
-                                       g_object_ref (task));
+  /* Save the expiry seconds. */
+  g_autoptr(GFile) expiry_seconds_file = get_expiry_seconds_file (self);
+  guint64 now_secs = epg_clock_get_time (self->clock);
+  if (now_secs > self->expiry_time_secs)
```

Comment:
There's a lot of `x > y ? 0 : y - x` in this diff (whether literally, or as here split across several statements). I suppose `MAX (0, y - x)` would be considered Very Bad in the case of underflow, though.